packages/medusa/src/services/claim-item.ts

```diff
@@ -1,13 +1,32 @@
 import { MedusaError } from "medusa-core-utils"
-import { BaseService } from "medusa-interfaces"
-
-class ClaimItemService extends BaseService {
+import { DeepPartial, EntityManager } from "typeorm"
+import { TransactionBaseService as BaseService } from "../interfaces"
+import { ClaimImage, ClaimItem, ClaimTag } from "../models"
+import { ClaimImageRepository } from "../repositories/claim-image"
+import { ClaimItemRepository } from "../repositories/claim-item"
+import { ClaimTagRepository } from "../repositories/claim-tag"
+import { CreateClaimInput, CreateClaimItemInput } from "../types/claim"
+import { FindConfig, Selector } from "../types/common"
+import { buildQuery, setMetadata, validateId } from "../utils"
+import EventBusService from "./event-bus"
+import LineItemService from "./line-item"
+
+class ClaimItemService extends BaseService<ClaimItemService> {
   static Events = {
     CREATED: "claim_item.created",
     UPDATED: "claim_item.updated",
     CANCELED: "claim_item.canceled",
   }
 
+  protected readonly lineItemService_: LineItemService
+  protected readonly eventBus_: EventBusService
+  protected readonly claimItemRepository_: typeof ClaimItemRepository
+  protected readonly claimTagRepository_: typeof ClaimTagRepository
+  protected readonly claimImageRepository_: typeof ClaimImageRepository
+
+  protected readonly manager_: EntityManager
```

Comment:
Issue: should not be readonly since it can be assigned out of the constructor

Reply:
Comment: also, the compiler should not be happy no? Since the base service require them to not be readonly

Reply:
in fact it is normal, https://github.com/microsoft/TypeScript/issues/14778, it still work since the assignation is done in the parent class and that in the parent class it is not readonly :) the child class has more capabilities compared to the parent class.